### PR TITLE
Update install instructions on documentation

### LINKED ISSUE #395: FR: Support complete installation via conda
#### Summary:
[conda](https://docs.conda.io/en/latest/index.html) is a package manager which specializes in creating Python environments which also require non-Python dependencies. This serves as a perfect fit for this kind of project, and would make installation (particularly on Windows) much easier without the need for something like `install_cxx_toolchain` or `install_cmdstan`, neither of which live 100% happily in a conda environment at the moment.

#### Description:
conda allows you to install both Python and other dependencies in a sandboxed way. This is particularly helpful on Windows, if the user doesn't want  mingw32's many binaries on their PATH.

Currently, you can install everything you need in a sandboxed way like this:

```powershell
conda create -n cmdstan python=3.9
conda activate cmdstan
# this is the equivalent of installing the toolchain, but it handles all environment variables (including PATH) for you
conda install -c msys2 m2-base m2-findutils m2w64-toolchain
# this seems to be unofficial? https://github.com/conda-forge/cmdstanpy-feedstock
conda install -c conda-forge cmdstanpy
# this is probably the least nice part, installing in a subfolder of the environment to keep it all tidy
python -m cmdstanpy.install_cmdstan -d "C:\ProgramData\Miniconda3\envs\cmdstan\cmdstan"
# this tells cmdstanpy where we installed it, and will be added or removed when the environment is activated/deactivated
conda env config vars set CMDSTAN="C:\ProgramData\Miniconda3\envs\cmdstan\cmdstan\cmdstan-2.27.0"
conda activate cmdstan
# good to go
```

Ultimately, I think it would be relatively easy to get this down to just
```powershell
conda create -n cmdstan python=3.9
conda activate cmdstan
conda install cmdstanpy
```
IF cmdstan itself had a conda build package created and the proper dependencies were all set in each package. I wasn't sure if this issue would belong here or in that repo, so I'm starting here.

#### Additional Information:
This seems like it was previously discussed in #7. I think that the `install_cxx_toolchain` command is useful for Windows folks using pip, but I see no reason why both couldn't be supported. I know this would be very helpful on Windows, users of other platforms may still find it simpler to have cmdstan installed as part of the environment process, and this would allow multiple versions to be used on the same machine side-by-side relatively easily. 

The current conda-forge feedstock for cmdstanpy is maintained by @maresb, who may have some thoughts on this

### PR DESCRIPTION
**Pull request by WardBrian**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and open-source license: see below

#### Summary

Update the installation instructions to favor conda installation.

Resolves #395 

Depends on ~https://github.com/conda-forge/staged-recipes/pull/15203~, https://github.com/conda-forge/cmdstanpy-feedstock/pull/4 hence the draft status

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):
Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Review by mitzimorris — APPROVED** [review]
LGTM

**Comment by mitzimorris** [comment]
ignore, approval - looks good so far - didn't notice the draft status.

File: docsrc/installation.rst
Comment (ahartikainen) [inline docsrc/installation.rst:21]: This is good line (one thing to remember is to call `-c conda-forge` when creating a new environment -> C++ stuff goes wrong if this is not done, but tbh this probably doesn't affect cmdstanpy use unless there is something clashing between user installed compiler and conda-forge compiled cmdstan)

File: docsrc/installation.rst
Comment (ahartikainen) [inline docsrc/installation.rst:46]:
```suggestion
* ``xarray``, an n-dimension labeled dataset package which can be used for outputs
```

File: docsrc/installation.rst
Comment (ahartikainen) [inline docsrc/installation.rst:68]: This is not true anymore (unless users use RTools to compile their models)

**Review by ahartikainen — COMMENTED** [review]
Added some comments

File: docsrc/installation.rst
Comment (WardBrian) [reply docsrc/installation.rst:21]: `-c conda-forge` is part of the command you highlighted - are you suggesting that this be called out in the instructions more specifically?

File: docsrc/installation.rst
Comment (WardBrian) [reply docsrc/installation.rst]: Does the updated text in the latest commit work, or should we just remove this paragraph?

File: docsrc/installation.rst
Comment (ahartikainen) [reply docsrc/installation.rst:21]: Not sure, but 50% of instructions get this correctly (by accident) and other half don't care.

**Comment by WardBrian** [comment]
https://github.com/conda-forge/staged-recipes/pull/15203 has been merged, so[ cmdstan is now available on conda-forge](https://anaconda.org/conda-forge/cmdstan)

**Comment by WardBrian** [comment]
@maresb - We're going to update the installation documentation for cmdstanpy once everything propagates in the conda-forge world. Do you intend to update cmdstanpy-feedstock to list the new `cmdstan` package as a dependency, or should we list both in our install command (something like `conda create -n stan -c conda-forge cmdstan cmdstanpy`)?

**Comment by maresb** [comment]
@WardBrian ya, I want to add the dependency, that's the reason I made the attempt to get cmdstan on cf. :smile: So either `conda create -n stan -c conda-forge cmdstan` or `conda create -n stan -c conda-forge cmdstanpy` should be sufficient, depending on intent.

I don't have time right at this very moment to test it, but I opened https://github.com/conda-forge/cmdstanpy-feedstock/pull/4 and added you as a maintainer, so you can push commits. (It should be more seamless than working from my `staged-recipes` repo.)

Do we need to update anything for the detection logic? It's been a while since I've looked into that.

Oh, I'm just seeing now that the build for my PR is failing. :( I don't have time right this moment, so feel free to take care of it, or I'll get to it eventually.

**Comment by WardBrian** [comment]
I'll take a look at it. The detection logic should be good to go - I set up the cmdstan build such that it sets `$CMDSTAN` when you activate the environment, which seems to work seamlessly with cmdstanpy (and hopefully cmdstanr as well, if users install cmdstan this way)

**Review by mitzimorris — APPROVED** [review]
looks good!

**Comment by WardBrian** [comment]
`conda install -c conda-forge cmdstanpy` now installs cmdstanpy and cmdstan, so this doc is ready to go

**Comment by WardBrian** [comment]
I added a small note about the use of `conda activate`. I'd like to go ahead and merge this, unless there are any more notes

**Comment by maresb** [comment]
In general regarding installation instructions with `conda`, I am in favor of mentioning [`mamba`](https://github.com/mamba-org/mamba) wherever possible. It's a drop-in replacement for `conda` which is orders of magnitude faster with large environments, yet somehow almost nobody knows of it.

**Comment by WardBrian** [comment]
I am hesitant to add many more conditional statements to the installation instructions (another ", or you can ___ if you ___" ). 

This new installation path is particularly targeting people who have difficulty installing the software currently, and I think (to overgeneralize) that kind of user will be better off with the solution that is easier to search online and generally more common

**Comment by maresb** [comment]
I understand. Up to you of course, but you might want to consider something minimally obtrusive like
```rst
If you use `conda <https://docs.conda.io/en/latest/>`__
or interchangeably `mamba <https://github.com/mamba-org/mamba>`__,
```

My motivation for pushing `mamba` is simply that `conda` [chokes in a catastrophic way](https://github.com/iterative/dvc.org/issues/2370#issuecomment-818891218) as soon as the number of dependencies starts growing. If we want people to use `conda` then I believe we need to raise awareness of the existence of a sustainable solution like `mamba`.

If you feel like it would be too distracting, then please skip it, I don't want to hold up your merge.

**Pull request merged**
